velox/functions/prestosql/HyperLogLogFunctions.h

```diff
@@ -78,4 +78,149 @@ struct EmptyApproxSetWithMaxErrorFunction {
     return true;
   }
 };
+
+class MergeHllFunction : public exec::VectorFunction {
+ public:
+  static std::vector<exec::FunctionSignaturePtr> signatures() {
+    return {exec::FunctionSignatureBuilder()
+                .returnType("hyperloglog")
+                .argumentType("array(hyperloglog)")
+                .build()};
+  }
+
+  void apply(
+      const SelectivityVector& rows,
+      std::vector<VectorPtr>& args,
+      const TypePtr& outputType,
+      exec::EvalCtx& context,
+      VectorPtr& result) const override {
+    VELOX_CHECK_EQ(args.size(), 1);
+
+    context.ensureWritable(rows, outputType, result);
+    auto* flatResult = result->as<FlatVector<StringView>>();
+
+    exec::LocalDecodedVector arrayDecoder(context, *args[0], rows);
+    auto decodedArray = arrayDecoder.get();
+    auto baseArray = decodedArray->base()->as<ArrayVector>();
+    auto rawSizes = baseArray->rawSizes();
+    auto rawOffsets = baseArray->rawOffsets();
+    auto indices = decodedArray->indices();
+    auto arrayElements = baseArray->elements();
+    auto flatArrayElements = arrayElements->as<FlatVector<StringView>>();
+
+    memory::MemoryPool* memoryPool = context.pool();
+    using SparseHll = common::hll::SparseHll<memory::MemoryPool>;
+    using DenseHll = common::hll::DenseHll<memory::MemoryPool>;
+
+    rows.applyToSelected([&](vector_size_t row) {
+      if (decodedArray->isNullAt(row)) {
+        flatResult->setNull(row, true);
+        return;
+      }
+
+      auto arraySize = rawSizes[indices[row]];
+      auto arrayOffset = rawOffsets[indices[row]];
+
+      int8_t indexBitLength = -1;
+      bool isSparse = true;
+      std::optional<SparseHll> sparseHll;
+      std::optional<DenseHll> denseHll;
+
+      for (vector_size_t i = 0; i < arraySize; ++i) {
+        auto elementIndex = arrayOffset + i;
+        if (arrayElements->isNullAt(elementIndex)) {
+          continue;
+        }
+
+        auto hllData = flatArrayElements->valueAt(elementIndex);
+        if (hllData.empty()) {
+          continue;
+        }
+
+        bool isSparseHllData = false;
+        bool isDenseHllData = false;
+
+        if (indexBitLength < 0) {
+          if (common::hll::DenseHlls::canDeserialize(hllData.data())) {
+            isDenseHllData = true;
```

Comment:
It might be cleaner to write this as follows.

```
const bool isSparseHllData =
              common::hll::SparseHlls::canDeserialize(hllData.data());
const boo isDenseHllData = !isSparseHllData &&
              common::hll::DenseHlls::canDeserialize(hllData.data());

if (indexBitLength < 0) {
   if (isSparseHllData) {
       indexBitLength =...
    } else {
       indexBitLength =...
    }
}
```